src/kibana/directives/saved_object_finder.js

```diff
@@ -104,100 +108,121 @@ define(function (require) {
           filterResults();
         });
 
-        $scope.selectedItem = false;
-        $input.on('keydown', (function () {
-          var enter = 13;
-          var up = 38;
-          var down = 40;
-          var left = 37;
-          var right = 39;
-          var esc = 27;
+        //manages the state of the keyboard selector
+        self.selector = {
+          enabled: false,
+          index: -1
+        };
 
-          var scrollIntoView = function ($el, snapTop) {
-            var el = $el[0];
+        //key handler for the filter text box
+        self.filterKeyDown = function ($event) {
+          if (keymap[$event.keyCode] !== 'tab') return;
 
-            if (!el) return;
+          if (self.hits.length === 0) return;
 
-            if ('scrollIntoViewIfNeeded' in el) {
-              el.scrollIntoViewIfNeeded(snapTop);
-            } else if ('scrollIntoView' in el) {
-              el.scrollIntoView(snapTop);
-            }
-          };
+          self.selector.index = 0;
+          self.selector.enabled = true;
 
-          return function (event) {
-            var $next;
-            var goingUp;
+          $timeout(function () {
+            $list.find('a:first').focus();
+          });
 
-            switch (event.keyCode) {
-            case enter:
-              if (!$selected) return;
+          $event.preventDefault();
+        };
 
-              // get the index of the selected element
-              var i = $list.find('li').index($selected);
+        //key handler for the list items
+        self.hitKeyDown = function ($event, page, paginate) {
+          switch (keymap[$event.keyCode]) {
+            case 'tab':
+              if (!self.selector.enabled) break;
 
-              // get the related hit item
-              var hit = $scope.hits[i];
+              $list.find('li.active a').focus();
 
-              if (!hit) return;
+              self.selector.index = -1;
+              self.selector.enabled = false;
 
-              // check if there is a url for this hit
-              var url = $scope.makeUrl(hit);
+              if ($event.shiftKey) {
+                $input.focus();
```

Comment:
Yes. This is the focus trigger for when the user does a shit-tab combo... Which instead of dropping out of arrow-key navigation, it reassigns the focus to the textbox.